test/react_client_test.dart

```diff
@@ -173,6 +173,47 @@ main() {
       expect(result, isNull);
     });
   });
+
+  group('registerComponent', () {
+    test('throws with printed error', () {
+      expect(() => react.registerComponent(() => ThrowsInDefaultPropsComponent()), throwsA(Error));
+      expect(() {
+        try {
+          react.registerComponent(() => ThrowsInDefaultPropsComponent());
+        } catch (_) {}
+      }, prints(contains('Error when registering Component:')));
+    });
+  });
+
+  group('registerComponent2', () {
+    test('throws with specific error when defaultProps throws', () {
+      expect(() => react.registerComponent2(() => ThrowsInDefaultPropsComponent2()), throwsA(Error));
+      expect(() {
+        try {
+          react.registerComponent2(() => ThrowsInDefaultPropsComponent2());
+        } catch (_) {}
+      }, prints(contains('Error when registering Component2 when getting defaultProps')));
+    });
+
+    test('throws with specific error when propTypes throws', () {
+      expect(() => react.registerComponent2(() => ThrowsInPropTypesComponent2()), throwsA(Error));
+      expect(() {
+        try {
+          react.registerComponent2(() => ThrowsInPropTypesComponent2());
+        } catch (_) {}
+      }, prints(contains('Error when registering Component2 when getting propTypes')));
+    });
+
+    test('throws with generic error when something else throws', () {
+      expect(() => react.registerComponent2(() => DartComponent2Component(), bridgeFactory: (component) => throw Error),
+          throwsA(Error));
```

Comment:
This is actually throwing the `Error` type instead of an actual error instance.

It works, but it's not really best practice and could confuse devs working on this code in the future.

Could we update this and other tests/`throw`s to not throw the types?

```suggestion
      expect(() => react.registerComponent2(() => DartComponent2Component(), bridgeFactory: (component) => throw StateError('bad bridge')),
          throwsStateError);
```